source/features/remove-unused-repo-tabs.tsx

```diff
@@ -0,0 +1,113 @@
+import cache from 'webext-storage-cache';
+import React from 'dom-chef';
+import select from 'select-dom';
+import elementReady from 'element-ready';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+import fetchDom from '../helpers/fetch-dom';
+import getTabCount from '../github-helpers/get-tab-count';
+import looseParseInt from '../helpers/loose-parse-int';
+import {getWorkflows} from './next-scheduled-github-action';
+import abbreviateNumber from '../helpers/abbreviate-number';
+import {onlyShowInDropdown} from './more-dropdown';
+import {buildRepoURL, getRepo} from '../github-helpers';
+
+function tabCannotBeHidden(tab: HTMLElement | undefined): boolean {
+	return (
+		!tab || // Tab disabled 🎉
+		tab.matches('.selected') ||// User is on tab 👀
+		// Repo/Organization owners should see the tab. If they don't need it, they should disable the feature altogether
+		pageDetect.canUserEditRepo() ||
+		pageDetect.canUserEditOrganization()
+	);
+}
+
+function setTabCounter(tab: HTMLElement, count: number): void {
+	const tabCounter = select('.Counter', tab)!;
+	tabCounter.textContent = abbreviateNumber(count);
+	tabCounter.title = count > 999 ? String(count) : '';
+}
+
+const getWikiPageCount = cache.function(async (): Promise<number> => {
+	const wikiPages = await fetchDom(buildRepoURL('wiki'), '#wiki-pages-box .Counter');
+	return looseParseInt(wikiPages);
+}, {
+	maxAge: {hours: 1},
+	staleWhileRevalidate: {days: 5},
+	cacheKey: () => 'wiki-page-count:' + getRepo()!.nameWithOwner
+});
+
+async function initWiki(): Promise<void | false> {
+	const wikiTab = await elementReady('[data-hotkey="g w"]');
+	if (!wikiTab) {
+		return false;
+	}
+
+	const wikiPageCount = await getWikiPageCount();
+	if (tabCannotBeHidden(wikiTab) || wikiPageCount > 0) {
+		setTabCounter(wikiTab, wikiPageCount);
+	} else {
+		onlyShowInDropdown('wiki-tab');
+	}
+}
+
+async function initActions(): Promise<void | false> {
+	const actionsTab = await elementReady('[data-hotkey="g a"]');
+	if (!actionsTab) {
+		return false;
+	}
+
+	const actionsCount = (await getWorkflows()).length;
+	if (tabCannotBeHidden(actionsTab) || actionsCount > 0) {
+		setTabCounter(actionsTab, actionsCount);
+	} else {
+		onlyShowInDropdown('actions-tab');
+	}
+}
+
+async function initProjects(): Promise<void | false> {
+	const projectsTab = await elementReady([
+		'[data-hotkey="g b"]', // In organizations and repos
+		'[aria-label="User profile"] [href$="?tab=projects"]' // In user profiles
+	].join());
+	if (tabCannotBeHidden(projectsTab) || await getTabCount(projectsTab!) > 0) {
+		return false;
+	}
+
+	if (pageDetect.isOrganizationProfile()) {
+		// https://github.com/orgs/USER/projects/new
+		const path = '/orgs' + location.pathname.split('/', 3).join('/');
+		select('.Header [href="/new"]')!.parentElement!.append(
+			<a className="dropdown-item" href={path}>
+				New project
+			</a>
+		);
+		projectsTab!.remove();
+		return;
+	}
```

Comment:
Should we just drop this altogether?

Reply:
I just answered thay

Reply:
Sorry I was not clear. My question was should we drop the whole feature on `pageDetect.isOrganizationProfile()`

Reply:
I think we should only hide the tabs if we're not owners/members, so this dropdown item should not exist in the first place.

Ownership is easy to detect through the Settings tab. Membership can be detected via the "New X" button on some pages:

<img width="1003" alt="Screen Shot 7" src="https://user-images.githubusercontent.com/1402241/124493370-c0fa3280-dddf-11eb-8c92-9237e3cd3a9c.png">


<img width="1030" alt="Screen Shot 8" src="https://user-images.githubusercontent.com/1402241/124493362-be97d880-dddf-11eb-83fe-f894f54e8fa3.png">


The logic should be something like:

```
if we're owner or members
	return
else
	hide tabs
```

This exact logic means that if we can't determine whether we're owners/members from the DOM, the tab will be hidden. This is fine.